Move permissions classes to 'api' module

therealtod opened the pull request:
Since permissions classes are mean to be shared among all the modules that will interact with the api, it seems more logical to move the accounts.permissions.py file to more "centralized" position like the /api folder.

therealtod committed:
Move permissions classes to 'api' module

Since permissions classes are mean to be shared among all the modules that will interact with the api, it seems more logical to move the accounts.permissions.py file to more "centralized" position like the /api folder.

ssaavedra commented on api/permissions.py:
```diff
@@ -13,6 +13,6 @@ def has_object_permission(self, request, view, obj):
 
 
-class IsProfileOwnerOrStaff(IsAdminOrSelf):
+class IsPlaceOwnerOrAdmin(IsAdminOrSelf):
```
Why the name change on the class? Shouldn't it be a bit more generic?

And instead of returning obj.user, if it is the _owner_ of the object, shouldn't that be an `.owner` attribute? (So that the very attribute name describes the relationship purpose)

therealtod replied:
Indeed I think I'm using the wrong approach. Probably it would make more sense to just eliminate that class, or call it "IsObjectUserorAdmin".
Then, we could subclass 'isAdminOrSelf' in a separare permissions.py file within the module that needs that specific subclass.

So for instance, in the forthcoming 'places' module I can create the IsPlaceOwnerOrAdmin(IsAdminOrSelf) class and implement it according to the structure of the models defined in that module

ssaavedra commented:
> more "centralized" position like the /api folder.

Instead of centralized, think _reusable_; which is one of the things Software Engineering is about :wink:

ssaavedra commented:
BTW, you have a typo in

> mean

I suppose you tried to write _meant_.

therealtod committed:
Move permissions classes to 'api' module

Since permissions classes are meant to be shared among all the modules that will interact with the api, it seems more logical to move the accounts.permissions.py file to more reusable position like the /api folder.

ssaavedra commented:
LGTM. Please squash your changes when committing :wink:

therealtod committed:
Merge branch 'master' into bugfix/move-permission-classes

therealtod commented:
I was asked to update this branch before merging, I need an approval once more.

therealtod commented:
Did I approve it on my own writing 'approval' ? o.O

ssaavedra commented:
It seems you didn't need so, but I have no idea why :-)

ssaavedra commented:
You can now remove the original branch.

ssaavedra commented:
As for subclassing, it could even be a better design choice (I think) if some of our models implemented some `is_owner(self, user)` check, and we just had one class `IsOwnerorAdmin` which checked whether the model had such a is_owner function, instead of writting so many similar permission classes.
